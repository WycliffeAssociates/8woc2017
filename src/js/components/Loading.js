import React, { Component } from 'react';
import styled from 'styled-components';
import loadingRingImg from '../../assets/images/loadingRing.svg';

class Loading extends Component {
  render() {
    const {height, marginTop} = this.props;
    return (

      <Container height= {height} marginTop = {marginTop} >
        <h1>{this.props.txt.loading}</h1>
        <img src={loadingRingImg} alt="Loading..." />
      </Container>
    );
  }
}


const Container = styled.div`
  display: flex;
  width: 100%;
  height: ${props => props.height? props.height: '60vh' };
  margin-top: ${props => props.marginTop? props.marginTop: 0};
  flex-direction: column;
  justify-content: center;
  align-items: center;

`;
Container.displayName = 'Container';

export default Loading;
