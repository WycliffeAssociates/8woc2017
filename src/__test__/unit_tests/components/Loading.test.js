/* global describe it expect*/
import React from 'react';
import Loading from '../../../js/components/Loading';
import {shallow} from 'enzyme';

const mockProps =  {
  txt: {
    loading: 'Loading...',
  },
};
describe('Loading Component Test Suite', () => {
  const wrapper = shallow(<Loading {...mockProps} />);

  it('should render the component successfully', () => {
    expect(wrapper.find('Container').length).toEqual(1);
  });

  it('should have correct text', () => {
    const loadingText = wrapper.find('h1');
    expect(loadingText.text()).toEqual('Loading...');
  });

  it('should have an image', () => {
    expect(wrapper.find('img'));
  });


});
