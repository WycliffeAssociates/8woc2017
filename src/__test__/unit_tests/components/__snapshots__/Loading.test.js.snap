// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component Test Suite renders correctly and matches snapshot 1`] = `
<div
  className="sc-bdVaJa erGDqw"
  height={undefined}
>
  <h1>
    Loading...
  </h1>
  <img
    alt="Loading..."
    src="http://opentranslationtools.org/static/images/loadingRing.svg"
  />
</div>
`;
